dallinger/frontend/static/scripts/dallinger2.js

```diff
@@ -66,7 +66,64 @@ var dallinger = (function () {
     _isUndefined: function (value) {
       return typeof value === 'undefined';
     }
+  };
 
+  /**
+   * ``dallinger.identity`` provides information about the participant.
+   * It has the following string properties:
+   *
+   * ``recruiter``     - Type of recruiter
+   *
+   * ``hitId``         - MTurk HIT Id
+   *
+   * ``workerId``      - MTurk Worker Id
+   *
+   * ``assignmentId``  - MTurk Assignment Id
+   *
+   * ``mode``          - Dallinger experiment mode
+   *
+   * ``participantId`` - Dallinger participant Id
+   *
+   * @namespace
+   */
+  dlgr.identity = {
+    get recruiter() { return dlgr.storage.get("recruiter"); },
+    set recruiter(value) { dlgr.storage.set("recruiter", value); },
+    get hitId() { return  dlgr.storage.get("hit_id"); },
+    set hitId(value) {  dlgr.storage.set("hit_id", value); },
+    get workerId() { return  dlgr.storage.get('worker_id'); },
+    set workerId(value) {  dlgr.storage.set('worker_id', value); },
+    get assignmentId() { return  dlgr.storage.get('assignment_id'); },
+    set assignmentId(value) {  dlgr.storage.set('assignment_id', value); },
+    get mode() { return  dlgr.storage.get('mode'); },
+    set mode(value) {  dlgr.storage.set('mode', value); },
+    get participantId() { return dlgr.storage.get('participant_id'); },
+    set participantId(value) { dlgr.storage.set('participant_id', value); },
+    get fingerprintHash() { return dlgr.storage.get('fingerprint_hash'); },
+    set fingerprintHash(value) { dlgr.storage.set('fingerprint_hash', value);},
+
+    serverFormat: function () {
```

Comment:
What's serverformat? And should `participantId` not be included in the `return`?

Reply:
This is the weakest part of this change, in my opinion. This was basically a replacement for `get_hit_params()` function. The experiment server routes use different names for the various values than the JS code (differences of snake_case vs camelCase, primarily), so the motivation for this function was to consolidate this translation in one place. The `identity` object also does the translation the other direction in `identity.initialize()`, so it made sense to manage the translation entirely in this object. This falls down a little bit because in other places (the call to `/worker_complete` for example), we assemble a complex URL bit by bit, so we wind up using the server-side keys directly anyway.

I didn't include  `participantId` -> `participant_id` because that isn't actually used.

I'm going to consider removing this function.

Reply:
Removed. It was only being called once, so I think it's better off inlined where used.